### DISCUSSION TITLE
PE653 controller conflicts

### DISCUSSION THREAD

**Discussion by bluemantwo** [tagyoureit/nodejs-poolController#685] [App and O/S Config Questions]
New install.  Got the controller and the dashboard software up and running.  However, the controller does not seem to find any devices.  I did a "xxd /dev/ttyUSB0" to confirm I am getting valid data.  My setup is 

- one Pentair VSP
- one Aquarite chlorinator
- one Intermatic PE653 pool controller that mimics the Intellicom

Works fine on some really old software (from 5 years ago!), but not on this version.  Any ideas?  
[start.txt](https://github.com/tagyoureit/nodejs-poolController/files/10573401/start.txt)
[poolConfig.txt](https://github.com/tagyoureit/nodejs-poolController/files/10573405/poolConfig.txt)
[poolState.txt](https://github.com/tagyoureit/nodejs-poolController/files/10573406/poolState.txt)

**Comment by rstrouse** [comment]
Yes press the gear menu and add your Chlorinator and pump.  The chlorinator you added is set to be controlled by an external controller.  When you add it set Nixie as the master.

**Comment by rstrouse** [comment]
Wait are you running a PE365?

**Reply by bluemantwo** [reply]
Yup!  Intermatic PE653 firmware v.39

**Reply by bluemantwo** [reply]
And I really should just abandon the PE653 and go entirely with njsPC software as the controller, but that might be a summer project.  

So for now, I don't want njsPC to control the pump, just monitor it for speed, watts, status.  But being able to control  the chlorinator and, down the road, the IntelliValves, would be helpful.

**Reply by rstrouse** [reply]
Hmmm... this may take some thought since the PE653 doesn't really report anything.  The only thing we see would be pump packets.  njsPC however would see that as an opportunity to effectively control the pump.  

If you have the RS485 hooked up can I get you to click on the hamburger, then select the Logging tab and click the Capture Replay button.  I do not need it to reload the config, just spit out a couple of minutes of data from RS485.  After a couple of minutes click the End Capture button and upload the zip file it downloads to your browser here.

Honestly, when you first installed this Nixie was not in existence to run schedules, manage circuits, control valves, or even tie the chlorinator to when the pool is actually running.  It would be a short hop and a relay board to completely eliminate the PE653 and get full control.  I know there was some firmware for that board running around out there back in the day to control it via zwave.  Is that what you are running?

**Reply by bluemantwo** [reply]
Yes, I have been able to gain full control of all aspects of the PE653 including pump speed, thermostats, timers, relays, etc. using z-wave.  I was lucky enough to have Intermatic install a special software on my system (Frank did it before he left Intermatic).  Everything the hand held remote does, I can do.   I use Russ's old software to provide updates of pump status (speed, watts, on/off) only.  But I also use the old software to control my Aquarite chlorinator, and want to add my IntelliValves when that gets supported.   So I want to avoid trying to control the pump, just read it's contents.  I was able to hook it up and it did briefly read the status, but I think it tries to control it which kicks it out of control by the PE653, and the pump turns off and then on again. Kinds of goes nuts.  

I have njsPC disconnected right now, but did capture the attached log if that helps.  I can hook it back up later and get more log data.  

The reason I want to move to the new software is 1) better integration with Home Assistant and 2) get ready for IntelliValve support if that happens.  

[consoleLog(2023-02-04_18-31-07).log](https://github.com/tagyoureit/nodejs-poolController/files/10609840/consoleLog.2023-02-04_18-31-07.log)

**Reply by bluemantwo** [reply]
Also, please don't spend much time on this if it requires any new coding.  Not worth it.  So few people use the PE653.  And I also planned to get rid of the PE653 when I upgrade my gas heater to a heat pump and reconfigure my whole system.  Only control relay I will need is for my spa blower, and I already have a 240V Insteon device that will do that.

**Reply by rstrouse** [reply]
This was not really what I was looking for.  I really need to see the traffic on the RS485 bus.  The replay output will give me everything I need and you don't even have to weed through the files on the pi.

https://github.com/tagyoureit/nodejs-poolController/wiki/How-to-capture-all-packets-for-issue-resolution

Also, here is a guide for hooking up various bits of equipment using the Nixie controller that is part of njsPC.  The hardware listed is just a suggestion.  There have been quite a few Nixie controllers built out there that blow away the capabilities of the PE635.

https://github.com/tagyoureit/nodejs-poolController/wiki/DIY-Standalone-Nixie-Pool-Controller

EDIT:  I may have some suggestions when I see the output.  We may be able to set the master to something other than Nixie.  Never had a pump that needs to be controlled externally before.

**Comment by bluemantwo** [comment]
Here is the replay file. As I recall, my RS485 assignments were:

pump: 96
PE653: 16
App: 33
Chlor: 2
[replay.zip](https://github.com/tagyoureit/nodejs-poolController/files/10612044/replay.zip)


And again, do not waste too much of your time on this.  Having the new software monitor my pump (and not control it) is just an interim step until I decommission my PE653 and go full njsPC in the coming year.
